test/actions/navigate-to-challenge.js

```diff
@@ -47,19 +45,14 @@ describe('navigateToChallenge action', () => {
 
       let nextState = reducer(initialState, navigateToChallenge(1, 2));
 
-      expect(nextState).toEqual(initialState.merge({
+      // by starting with nextState and merging in what we expect,
+      // we eliminate all other properties from consideration
+      expect(nextState).toEqual(nextState.merge({
```

Comment:
I'm not sure about this. It seems like an easy way to have unintentional changes to the state be missed by the test. If the reducer changes a different property other than the ones listed here, the test will still pass.

I think it's ok to have explicit properties that we know will change and we don't care about their values (in this case we know that `challengeProgress` etc will change due to the navigation), but we don't want to miss other state changes.